### DISCUSSION TITLE
Pyright analyses whole project

### DISCUSSION THREAD

**Discussion by muke101** [microsoft/pyright#2091] [Q&A] [answered]
I used to get a prompt on opening a large python project asking if I wanted pyright to analyse all files, and that doing so may slow things down. I think I accidentally pressed 'y' once because it's not showing up on startup anymore and now my text editor (emacs) has periodic freezes where it didn't before. I've tried to place a `pyrightconfig.json` file at the root of the project with the following:

> {
>     "phython.analysis.diagnosticMode":"openFilesOnly"
> }

But this hasn't helped performance/freezes. How can I get pyright to only looking at a subsection of the project again, rather than the whole thing?

**Comment by jakebailey** [comment, marked as answer]
That's an editor setting and can't be set in a pyrightconfig.json (pyright itself as a CLI will always typecheck the project or specified files; an editor has different rules about open files versus the workspace).

You'd need to set that wherever your editor declares LSP settings. I'd also think that's where your editor stored that configuration you accepted.

**Comment by erictraut** [comment]
I'm curious which editor you're using. Pyright doesn't present any prompt like the one you're describing, so some other piece of code (either the editor or the LSP plugin for the editor) is presenting that prompt.

**Reply by muke101** [reply]
I'm using doom emacs.
